driver/src/main/java/org/neo4j/driver/internal/net/BoltServerAddress.java

```diff
@@ -138,21 +141,12 @@ public int port()
 
     /**
      * Determine whether or not this address refers to the local machine. This
-     * will generally be true for "localhost" or "127.x.x.x".
+     * will be true for "localhost" or "127.x.x.x".
      *
      * @return true if local, false otherwise
      */
     public boolean isLocal()
     {
-        try
-        {
-            // confirmed to work as desired with both "localhost" and "127.x.x.x"
-            return InetAddress.getByName( host ).isLoopbackAddress();
-        }
-        catch ( UnknownHostException e )
-        {
-            // if it's unknown, it's not local so we can safely return false
-            return false;
-        }
+        return LOCALHOST.matcher( host ).matches();
```

Comment:
Because this behaviour can be implemented uniformly across all different drivers on all different platforms, such that we have the same behaviour in all drivers.